tests/src/test/java/alluxio/IntegrationTestUtils.java

```diff
@@ -77,5 +85,55 @@ public static String randomString() {
     return sb.toString();
   }
 
+  /**
+   * Triggers two heartbeats to wait for a given list of blocks to be removed from both master and
+   * worker.
+   * Blocks until the master and block are in sync with the state of the blocks.
+   *
+   * @param bw the block worker that will remove the blocks
+   * @param blockIds a list of blockIds to be removed
+   */
+  public static void waitForBlocksToBeRemoved(final BlockWorker bw, long... blockIds) {
+    try {
+      // Schedule 1st heartbeat from worker.
+      Assert.assertTrue(HeartbeatScheduler.await(HeartbeatContext.WORKER_BLOCK_SYNC, 5,
+          TimeUnit.SECONDS));
+      HeartbeatScheduler.schedule(HeartbeatContext.WORKER_BLOCK_SYNC);
+
+      // Waiting for the removal of blockMeta from worker.
+      for (final long blockId : blockIds) {
+        CommonTestUtils.waitFor(new Function<Void, Boolean>() {
+          @Override
+          public Boolean apply(Void input) {
+            boolean hasBlockMeta = bw.hasBlockMeta(blockId);
+            // Need to wait for the remover actually finishes
+            if (!hasBlockMeta) {
+              BlockMasterSync sync = Whitebox.getInternalState(bw, "mBlockMasterSync");
```

Comment:
is there a way to wait here without using Whitebox, maybe waiting for `blockInfo.getLocations().isEmpty()` instead?